### ISSUE TITLE
Ability to use StatusLight in a Picker

### ISSUE THREAD

**Issue by schenglooi** [adobe/react-spectrum#4134]
# 🙋 Feature Request

The ability to use the `StatusLight` component as an `Item` in a `Picker` component

## 🤔 Expected Behavior

- Selected `StatusLight` component should be shown in the `Picker` as styled in the `Item` when the list is collapsed
- The `StatusLight` component should be rendered correctly as an `Item` in the list when the list is opened, i.e. the dot and text should be aligned properly.
- The `Picker` width should be resizable without causing the text part of the `StatusLight` to wrap
- If the `StatusLight` text is too long, should have the option to truncate with an ellipsis.

## 😯 Current Behavior

- The text part of the StatusLight wraps if the size is set to anything less than "`size-1400`"
- Picker width greater than "`size-1400`" does not seem to have an effect on the size of the Picker.
- The selection highlight for non-selected `Item` is incorrect when the list is open

Workaround for selection highlight:
```
<Text>
   <View>
        <StatusLight ... />
   <View>
<Text>
```

## 🔦 Context

This Picker component is intended to show progress. Our list has two items, "In Progress" and "Approved". The user chooses "Approved" to move something forward. The StatusLight was also used in a different view to indicate the "progress" for that item.

I ended up spending some time trying to implement workarounds using other components to duplicate StatusLight. All of which did not work out.

The result of the current implementation:
Note that the width of the Picker is too wide compared to the design.
<img width="327" alt="image" src="https://user-images.githubusercontent.com/26752761/222020188-a38862e3-c8cf-4c07-aff7-6d5a33eff5c9.png">

## 💻 Examples
mockup in dark mode;
![image](https://user-images.githubusercontent.com/26752761/222020862-07766e70-7e64-4f59-9aa4-3dc7da03c3fd.png)

## 🧢 Your Company/Team
Adobe/3D&I

3D&I, WF

**Comment by devongovett** [comment]
I think the solution to this should be an option for StatusLight rather than something specific to Picker. I think if StatusLight was `inline-flex` rather than `flex`, and removed its builtin padding (which is problematic in many scenarios), the issue would be solved. The question at this point is how to make those changes in a non-breaking way. We'll either need a new prop, a new component, or a new major version for this package.

**Comment by reidbarber** [comment]
Is there a downside to just fixing the style issues via slots?

**Comment by devongovett** [comment]
I don't think we should really do component specific overrides that way if we can help it. The problem isn't unique to Picker - the same padding also affects table cells and lots of other things. e.g. #2629

**Comment by devongovett** [comment]
We will address this use case in Spectrum 2.
